test/TokenRequestTest.js

```diff
@@ -1,51 +1,304 @@
+      from: rootAccount,
+    })
+    vault = await Vault.at(deployedContract(newVaultAppReceipt))
 
-contract('TokenRequest', ([rootAccount, vault, ...accounts]) => {
-    let daoDeployment = new DaoDeployment()
-    let tokenRequestBase, tokenRequest, tokenManagerBase, tokenManager
-    let FINALISE_TOKEN_REQUEST_ROLE, MINT_ROLE
+    await vault.initialize()
+    await tokenManager.initialize(requestableToken.address, false, 0)
+    await tokenRequest.initialize(tokenManager.address, vault.address)
 
-    before(async () => {
-        await daoDeployment.deployBefore()
+    mockErc20 = await MockErc20.new(rootAccount, MOCK_TOKEN_BALANCE)
+    await mockErc20.transfer(rootAccount, ROOT_TOKEN_AMOUNT)
+  })
 
-        tokenRequestBase = await TokenRequest.new()
-        FINALISE_TOKEN_REQUEST_ROLE = await tokenRequestBase.FINALISE_TOKEN_REQUEST_ROLE()
+  describe('initialize(address _tokenManager, address _vault)', () => {
+    it('sets correct variables', async () => {
+      const actualTokenManager = await tokenRequest.tokenManager()
+      const actualVault = await tokenRequest.vault()
 
-        tokenManagerBase = await TokenManager.new()
-        MINT_ROLE = await tokenManagerBase.MINT_ROLE()
+      assert.strictEqual(actualTokenManager, tokenManager.address)
+      assert.strictEqual(actualVault, vault.address)
     })
+  })
 
+  describe('function setTokenManager(address _tokenManager)', () => {
     beforeEach(async () => {
-        await daoDeployment.deployBeforeEach(rootAccount)
-        const newTokenRequestAppReceipt = await daoDeployment.kernel
-            .newAppInstance('0x1234', tokenRequestBase.address, '0x', false, {from: rootAccount})
-        tokenRequest = await TokenRequest.at(deployedContract(newTokenRequestAppReceipt))
+      const newTokenManagerAppReceipt2 = await daoDeployment.kernel.newAppInstance(
+        '0x5555',
+        tokenManagerBase.address,
+        '0x',
+        false,
+        { from: rootAccount }
+      )
+      tokenManager2 = await TokenManager.at(deployedContract(newTokenManagerAppReceipt2))
+    })
+    it('sets a token manager', async () => {
+      await daoDeployment.acl.createPermission(accounts[1], tokenRequest.address, SET_TOKEN_MANAGER_ROLE, rootAccount)
+
+      await tokenRequest.setTokenManager(tokenManager2.address, { from: accounts[1] })
+
+      const actualTokenManager = await tokenRequest.tokenManager()
+      assert.strictEqual(actualTokenManager, tokenManager2.address)
+    })
+  })
+
+  describe('function setVault(address _vault)', () => {
+    beforeEach(async () => {
+      const newVaultAppReceipt2 = await daoDeployment.kernel.newAppInstance('0x8889', vaultBase.address, '0x', false, {
+        from: rootAccount,
+      })
+      vault2 = await Vault.at(deployedContract(newVaultAppReceipt2))
+    })
+    it('sets a vault', async () => {
+      await daoDeployment.acl.createPermission(accounts[1], tokenRequest.address, SET_VAULT_ROLE, rootAccount)
```

Comment:
The createPermission() could go in the beforeEach hook so if we add more tests for this describe it's not necessary to create it in each test